charts/redpanda/service_internal.go

```diff
@@ -0,0 +1,141 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +gotohelm:filename=service.internal.go.tpl
+package redpanda
+
+import (
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func MonitoringEnabledLabel(dot *helmette.Dot) map[string]string {
+	values := helmette.Unwrap[Values](dot.Values)
+	if values.Monitoring.Enabled {
+		return map[string]string{
+			// no gotohelm support for strconv.FormatBool
+			"monitoring.redpanda.com/enabled": "true",
+		}
+	}
+	return map[string]string{
+		"monitoring.redpanda.com/enabled": "false",
+	}
+}
+
+func ServiceInternal(dot *helmette.Dot) *corev1.Service {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	ports := []corev1.ServicePort{}
+	if values.Listeners.Admin != nil {
```

Comment:
I've been meaning to update the typing on this but all members of the `Listeners` struct are always present. They just accidentally got added as pointers. Want to update the types and remove the nil checks?

Reply:
Can do. I added the nil checks when I saw regression tests crapping out but either work.